scripts/add_application_version.py

```diff
@@ -0,0 +1,85 @@
+from sqlalchemy import create_engine, MetaData
+from sqlalchemy.exc import SQLAlchemyError
+import csv
+from datetime import datetime
+from argparse import ArgumentParser
+import logging
+import sys
+
+logging.basicConfig(level=logging.INFO)
+
+DESC= """Scrpt to load new application versions to status-db"""
+
+VALID_HEADERS = ['App tag', 'Version', 'Valid from', 'Standard', 'Priority', 'Express', 'Research', 'Comment']
+
+def add_application_version(file_path, conection_string, sign):
+    """
+    Args:
+        file_path:          Path to csv file with heathers: 'App tag', 'Valid from', 'Standard', 
+                            'Priority', 'Express', 'Research'
+        connection_string: 'mysql+pymysql://<user>:<password>@<hoast>:<port>/cg'
+        """
+    try:
+        engine = create_engine(conection_string)
+        connection = engine.connect() 
+        metadata = MetaData()
+        metadata.reflect(bind=engine)   
+        table = metadata.tables['application_version']
+    except SQLAlchemyError as e:
+        sys.exit('Failed to connect: %s' % (e))
+
+    try:
+        f = open(file_path)
+        rows=csv.DictReader(f, delimiter=',')
+    except IOError as e:
+        sys.exit('Failed read file: %s' % (e))
+
+    if not len(set(rows.fieldnames))==len(rows.fieldnames):
+        sys.exit('Heades in file are not unique')
+
+    if not set(rows.fieldnames) < set(VALID_HEADERS):
+        sys.exit('Headers in file are not valid. Should contain the following: %s' % (str(VALID_HEADERS)))
+    
+
+    for row in rows:
+        tag = row['App tag']
+        query = f'select max(application_version.version), application.tag, application.id from application_version inner join application on application_version.application_id=application.id where application.tag="{tag}"'
+        latest_version, app_tag, app_id = engine.execute(query).first()
+        if not latest_version:
+            latest_version=0
+        if app_tag:
```

Comment:
what if there is no app_tag? I think it should not pass by silently.
I suggest we abort execution and rollback the whole price update,  what do you think?

Reply:
Good point. I wont havt time to fix this before leaving today and Im on vacation next week. Feel free to contribute!

Reply:
I changed the script to check for all applications first and exit if any is missing so that the user can deal with the problems before script is allowed to change the database